pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SnapshotSegmentAbortedTxnProcessorImpl.java

```diff
@@ -0,0 +1,758 @@
+                index.setAbortedMarkEntryID(abortedMarkerPersistentPosition.getEntryId());
+                index.setSegmentLedgerID(((MessageIdImpl) messageId).getLedgerId());
+                index.setSegmentEntryID(((MessageIdImpl) messageId).getEntryId());
+
+                indexes.put(abortedMarkerPersistentPosition, index);
+                //update snapshot segment index.
+                //If the index can not be written successfully, the snapshot segment wil be overwritten
+                // when the processor writes snapshot segment next time.
+                return updateSnapshotIndex(new TransactionBufferSnapshotIndexesMetadata(
+                                maxReadPosition.getLedgerId(), maxReadPosition.getEntryId(), new LinkedList<>()),
+                        indexes.values().stream().toList());
+            });
+        }
+
+        // update index after delete all segment.
+        private CompletableFuture<Void> deleteSnapshotSegment(List<PositionImpl> positionNeedToDeletes) {
+            List<CompletableFuture<Void>> results = new ArrayList<>();
+            for (PositionImpl positionNeedToDelete : positionNeedToDeletes) {
+                long sequenceIdNeedToDelete = indexes.get(positionNeedToDelete).getSequenceID();
+                results.add(snapshotSegmentsWriterFuture
+                        .thenCompose(writer -> writer.deleteAsync(buildKey(sequenceIdNeedToDelete), null))
+                        .thenRun(() -> {
+                            if (log.isDebugEnabled()) {
+                                log.debug("[{}] Successes to delete the snapshot segment, "
+                                                + "whose sequenceId is [{}] and maxReadPosition is [{}]",
+                                        this.topic.getName(), this.sequenceID, positionNeedToDelete);
+                            }
+                            //The process will check whether the snapshot segment is null,
+                            // and update index when recovered.
+                            indexes.remove(positionNeedToDelete);
+                        }).exceptionally(e -> {
+                            log.warn("[{}] Failed to delete the snapshot segment, "
+                                            + "whose sequenceId is [{}] and maxReadPosition is [{}]",
+                                    this.topic.getName(), this.sequenceID, positionNeedToDelete, e);
+                            return null;
+                        }));
+            }
+            return FutureUtil.waitForAll(results);
+        }
+
+        private CompletableFuture<Void> updateSnapshotIndex(TransactionBufferSnapshotIndexesMetadata snapshotSegment,
+                                                            List<TransactionBufferSnapshotIndex> indexList) {
+            TransactionBufferSnapshotIndexes snapshotIndexes = new TransactionBufferSnapshotIndexes();
```

Comment:
the arg `List<TransactionBufferSnapshotIndex> indexList` of method `updateSnapshotIndex` is unnecessary, right?

Reply:
Yeap, fixed.